### PR TITLE
something exists in state

### PR DESCRIPTION
**Pull request by Peter-Springer**
adds fake Denver weather file that sets state

### CODE REVIEW

**Comment by Jeff-Duke** [comment]
Paired while writing

File: src/actions/index.js [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
-//       .then(json => dispatch({
-//         type: RECEIVE_FORECAST,
-//         json
-//       }));
-//   };
-// }
+export function fetchWeatherByZip(zip) {
+  return dispatch => {
+    dispatch({ type: REQUEST_FORECAST });
+    return fetch(API_URL.replace(/zip/, zip))
```
Comment (stevekinney) [inline src/actions/index.js:11]: This is likely to be slow and a bit brittle. What if we used a function that returned the string for us?

For example:

``` js
const apiUrl = (zip = '80202') => (
  `http://api.openweathermap.org/data/2.5/weather?zip=${zip},us&appid=b248d23bebc3f14165312ce58845f0c4`
);
```

Now, it's a super flexible function that returns a string that you can use.

``` js
fetch(apiUrl(zip)).then(…)
```

File: src/actions/index.js [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
+      .then(req => req.json())
+      .then(json => dispatch({
+        type: RECEIVE_FORECAST,
+        json
+      }));
+  };
+}
 
 // export const receiveForecast = (forecastData) => ({
```
Comment (stevekinney) [inline src/actions/index.js:20]: Master is sacred and no place for commented out code! 😛

File: src/data/weather.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+const denverWeather = {
```
Comment (stevekinney) [inline src/data/weather.js:1]: I'm going to guess this is a stub for now? That's totally cool, but I might consider denoting it as such.

File: src/index.js [hunk 4]
```diff
@@ -1,21 +1,26 @@
...
 import { Provider } from 'react-redux';
 import { browserHistory } from 'react-router';
 import Routes from './routes';
 import thunk from 'redux-thunk';
 import createLogger from 'redux-logger';
-import weather from './reducers/weather';
+import rootReducer from './reducers/rootReducer';
+import denverWeather from './data/weather'
 require("!style!css!sass!./style/styles.scss");
+console.log(denverWeather)
```
Comment (stevekinney) [inline src/index.js:12]: This probably shouldn't make it into master.

File: src/reducers/rootReducer.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { combineReducers } from 'redux'
+import weather from './weather'
+
+const rootReducer = combineReducers({
+  weather
```
Comment (stevekinney) [inline src/reducers/rootReducer.js:5]: Do we need to combine reducers if we only have one reducer?

File: src/reducers/weather.js [hunk 6]
```diff
@@ -2,10 +2,10 @@ import { RECEIVE_FORECAST, REQUEST_FORECAST } from '../actions/index.js';
 
 const weather = (state = {}, action) => {
   switch (action.type) {
-    case REQUEST_FORECAST:
+    case 'REQUEST_FORECAST':
```
Comment (stevekinney) [inline src/reducers/weather.js:5]: I thought the constants we pretty clever. 😞

**Pull request merged**